babel/localtime/_unix.py

```diff
@@ -65,17 +65,21 @@ def _get_localzone(_root='/'):
     # since it knows where the zone files are that should be a bit
     # better than reimplementing the logic here.
     if sys.platform == 'darwin':
-        c = subprocess.Popen(['systemsetup', '-gettimezone'],
-                             stdout=subprocess.PIPE)
-        sys_result = c.communicate()[0]
-        c.wait()
-        tz_match = _systemconfig_tz.search(sys_result)
-        if tz_match is not None:
-            zone_name = tz_match.group(1)
-            try:
-                return pytz.timezone(zone_name)
-            except pytz.UnknownTimeZoneError:
-                pass
+        try:
+            c = subprocess.Popen(['systemsetup', '-gettimezone'],
+                                stdout=subprocess.PIPE)
+            sys_result = c.communicate()[0]
+            c.wait()
+            tz_match = _systemconfig_tz.search(sys_result)
+            if tz_match is not None:
+                zone_name = tz_match.group(1)
+                try:
+                    return pytz.timezone(zone_name)
+                except pytz.UnknownTimeZoneError:
+                    pass
+        # iOS doesn't come with systemsetup
+        except FileNotFoundError: 
+            pass
```

Comment:
Instead of having a huge `try`-block, you can make it more specific moving the non-critical things to the `else`-block of a try-except:
```suggestion
        try:
            c = subprocess.Popen(['systemsetup', '-gettimezone'],
                                stdout=subprocess.PIPE)
        # iOS doesn't come with systemsetup
        except FileNotFoundError: 
            pass
        else:
            sys_result = c.communicate()[0]
            c.wait()
            tz_match = _systemconfig_tz.search(sys_result)
            if tz_match is not None:
                zone_name = tz_match.group(1)
                try:
                    return pytz.timezone(zone_name)
                except pytz.UnknownTimeZoneError:
                    pass        
```
it helps identifying which part we expect to throw the exception, and avoids catching any unexpected instances of this exceptions – which probably should stay unhandled so that the bug becomes visible and one gets a nice backtrace.

_(not a maiintainer, just randomly browsed through PRs)_